src/mds/MDSAuthCaps.cc

```diff
@@ -356,8 +356,8 @@ bool MDSAuthCaps::parse(string_view str, ostream *err)
 
     if (err) {
       if (string(iter, end).find("allow") != string::npos) {
-       *err << "Permission flags in MDS caps must start with 'r' or " <<
-	       "'rw' or be '*' or 'all'";
+       *err << "Permission flags in MDS capability string must either start "
```

Comment:
nit: `either` can be dropped IMO.

Reply:
I added it deliberately because 4 `or`s in a sentence wasn't sounding great.

Reply:
How about: `Invalid permission flag - allowed permissions are: r, rw....`

?

Reply:
There are more than few combinations to list, so it'll be confusing for users.

Reply:
"r or rw or be * or all" sounds a bit cloudy.

Maybe: Permission flags in MDS caps must start with 'r' (e.g., r, rw, etc..) or */all.
 WDYT?

Reply:
> "r or rw or be * or all" sounds a bit cloudy.

Agreed. I wish it could be more elegant.

> Maybe: Permission flags in MDS caps must start with 'r' (e.g., r, rw, etc..) or */all.

Clearer but too many symbols.

What about `Permission flags in MDS caps must be '*' or 'all' or must start with 'r'`?

Reply:
sounds fine.

Reply:
Done.